src/embind/embind.js

```diff
@@ -560,10 +561,10 @@ var LibraryEmbind = {
         var base = _malloc({{{ POINTER_SIZE }}} + length + 1);
         var ptr = base + {{{ POINTER_SIZE }}};
         {{{ makeSetValue('base', '0', 'length', SIZE_TYPE) }}};
-        if (stdStringIsUTF8 && valueIsOfTypeString) {
-          stringToUTF8(value, ptr, length + 1);
-        } else {
-          if (valueIsOfTypeString) {
+        if (valueIsOfTypeString) {
+          if (stdStringIsUTF8) {
+            stringToUTF8(value, ptr, length);
```

Comment:
Looks like tests are still failing, `+ 1` is needed or something else going on?

Reply:
Yup, you are correct.

Strange.. his code doesn't actually require null termination (it uses lenght-pefix instead)... but stringToUTF8 unconditionally does.